napari/_vispy/vendored/volume.py

```diff
@@ -959,6 +966,54 @@ def _in_loop_snippet(self):
     def _after_loop_snippet(self):
         return frag_dict[self.method]['after_loop']
 
+    @staticmethod
+    @lru_cache(maxsize=10)
```

Comment:
out of interest, do you have a sense for how performant this is? Does it scale well with the number of clipping planes?

Reply:
Do you mean how performant caching is, or how performant many clipping planes are? I expect caching to be negligible, and the number of clipping planes to be more important. However, as mentioned upstream, it's probably fair to assume clipping planes won't be used in amounts bigger than ~10, at which point you can probably merge 2 almost-equal planes into 1 and not even notice.

Reply:
I meant the performance of the clipping planes in general! I can actually imagine us using this relatively heavily - given that we may use 6 planes for a bounding box and may want to add more... can cross that bridge when we come to it though :)

Reply:
6 for a box + maybe 4 for additional clipping is 10 :P 2 boxes seems already redundant... But yeah, I don't expect this to be problematic any time soon!